doc/architecture-quick-reference.rst

```diff
@@ -0,0 +1,468 @@
+Architecture Quick Reference
+============================
+
+This is a quick reference guide to IoTempower's architecture and key concepts.
+
+For detailed information, see:
+
+- `Architecture Overview <architecture.rst>`_ - Complete system architecture
+- `Device Architecture <device-architecture.rst>`_ - Device framework details
+- `Deployment Process <deployment-process.rst>`_ - Build and deploy workflow
+
+
+Key Concepts
+------------
+
+Systems, Nodes, and Devices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   IoT System (e.g., "my-home")
+   └── Nodes (e.g., "living-room", "bedroom")
+       └── Devices (e.g., "temperature", "motion", "led")
+
+**System**: A complete IoT installation (one WiFi network, one MQTT broker)
+
+**Node**: A microcontroller (ESP8266/ESP32) with connected devices
+
+**Device**: A sensor or actuator (temperature sensor, LED, button, etc.)
+
+
+Device Class Hierarchy
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   Device (base class)
+   ├── Input_Base → Input, Analog
+   ├── Output → PWM, RGB_Single, RGB_Strip_Bus
+   ├── I2C_Device → BMP180, BH1750, etc.
+   ├── Dht11, Dht22, Ds18b20 (OneWire devices)
+   ├── Hcsr04 (acoustic distance sensor)
+   └── RGB_Base → RGB_Single, RGB_Strip_Bus
+
+Each level adds functionality:
+
+1. **Device**: Core lifecycle, MQTT, polling
+2. **Input_Base**: Pin management, buffers
+3. **Input**: Digital reading, debouncing
+
+
+Configuration Files
+-------------------
+
+system.conf (System Level)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   # WiFi and network
+   IOTEMPOWER_AP_NAME="my-iot-network"
+   IOTEMPOWER_AP_PASSWORD="secret123"
+   
+   # MQTT broker
+   MQTT_HOST="192.168.1.100"
+   MQTT_USER="mqtt"
+   MQTT_PASSWORD="mqttpass"
+
+node.conf (Node Level)
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   # Board type - any board listed in lib/node_types/ can be used
+   board="wemos_d1_mini"
+   # or: board="esp32dev", "nodemcu", "esp32", "lolin_s2_mini", etc.
```

Comment:
add also m5stickc and m5stickcplus here as well as esp32minikit, sonoff, and lolin_s2_mini to the commented examples - skip the etc.